### PR TITLE
Add tutorial for lumbar segmentation and registration

### LINKED ISSUE #3599: Provide solutions for registration of lumbar data
We should provide a satisfactory solution for every users working with thoraco/lumbar data. Currently, registration to the template is not easy.

Subtasks:
- [ ] https://github.com/spinalcordtoolbox/PAM50/issues/26
- [x] https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4180

Ref:
- https://forum.spinalcordmri.org/t/labels-for-lumbar-spinal-level/737/5

### LINKED ISSUE #4180: Add SCT tutorial about registering lumbar images using 2 labels
This is a feature request that came out of the SCT dev meeting on 2023-07-27 ([minutes](https://docs.google.com/document/d/1ItApJQfajO2lRzOU2yenWbeRg6alfsdut3J4AVVdo78/edit#heading=h.3jazj3wbpm6z)), motivated by several discussions at recent conferences.

We should add explicit tips in SCT tutorials and courses about registration of lumbar images using 2 labels. There's already an explanation of this in a forum post ([link](https://forum.spinalcordmri.org/t/labels-for-lumbar-spinal-level/737/8?u=valosekj)), but most people don't see it (understandably!). Maybe we should also update the usage string for `sct_register_to_template`?

For context:
- Possibly related to #2607 and #3073 about running `sct_label_vertebrae` in the lumbar FOV?
- For using **more than 2** labels, this is related to the documentation update in #3602.
- For the lumbar SC segmentation of T2w images, we have a deepseg model since #3766.
- This is _almost_ a duplicate of #3599, which is about the same problem, but from the technical side, rather than the documentation/training side.

### PR DESCRIPTION
**Pull request by joshuacwnewton**
## Description

This tutorial compiles the recommendations from the forum posts into a web tutorial. A [corresponding section](https://docs.google.com/presentation/d/1t40Fd0fS0SwWR5FU_GWKEvHkB9d96LVddLQW6L3QAx0/edit#slide=id.g24c79a1b6f2_0_152) has also been added to the SCT Course slides.

> Note: You will need to reinstall the PAM50 template from this branch in order to get the necessary label `60` for registration.

You can find the tutorial for this PR here: https://spinalcordtoolbox--4250.org.readthedocs.build/user_section/tutorials/registration-to-template/lumbar-registration.html

## Linked issues

Fixes #4180.
Fixes #3599 (once https://github.com/spinalcordtoolbox/PAM50/pull/27 is merged, anyway!)

### CODE REVIEW

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+Deciding which landmarks to use (discs, conus medullaris)
+=========================================================
+
+Registration can be performed with either one or two labels. So, given the tradeoffs between the aligning the discs and/or the conus medullaris, the question becomes: Should registration be done with 1 disc label, 1 conus medullaris label, or a combination of the two?
+
+.. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/lumbar-registration/label-comparison.png
+   :align: center
+
+Above is a comparison of sample results, demonstrating that registration will vary significantly depending on which landmarks are chosen. The "ideal result" will depend on the goals of your research. However, in most cases, you will want to stick with one approach and use it consistently throughout your research.
```
Comment (joshuacwnewton) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst:20]: I'm not sure which of the registration results are "ideal".

@jcohenadad initially recommended just the cauda equinea label initially, but then we had a second user choose both the cauda equinea _plus_ another landmark higher up in the cord. So, I included both techniques here for comparison.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+To apply the registration algorithm, the following command is used:
+
+.. code:: sh
+
+   sct_register_to_template -i t2_crop.nii.gz \
+                            -s t2_crop_seg.nii.gz \
+                            -ldisc t2_crop_labels.nii.gz \
+                            -c t2 -qc ~/qc_singleSubj \
+                            -param step=1,type=seg,algo=centermassrot:step=2,type=seg,algo=bsplinesyn,metric=MeanSquares,iter=3,slicewise=0:step=3,type=im,algo=syn,metric=CC,iter=3,slicewise=0
```
Comment (joshuacwnewton) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:14]: @jcohenadad What methodology did you use to select these parameters when [responding on the forum](https://forum.spinalcordmri.org/t/labels-for-lumbar-spinal-level/737/8?u=joshuacwnewton)? 

Should we repeat these parameters as-is, or should we start out with the default parameters (and then instruct the user to tweak parameters for their data and/or post on the forum)?

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Before starting this tutorial
+#############################
+
+#. Make sure that you have the following files in your working directory:
+
+   * ``single_subject/data/t2_lumbar/t2.nii.gz`` : An anatomical spinal cord scan in the T2 contrast showing visible lumbar regions.
```
Comment (joshuacwnewton) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst:6]: I'm starting to question the idea of using separate subfolders when dealing with `t2_lumbar` and `t2_compression` tutorials. If these tutorials get integrated into the SCT course, for example, it's a bit awkward to have to `cd` from `t2/` into these other folders and then back again... :thinking: 

I'd like to test and see how the flow of the tutorials works in practice? (Are we even including lumbar/compression tutorials in the course?)

**Review by joshuacwnewton — COMMENTED** [review]
Small self-review to highlight some areas that need discussion.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+.. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/lumbar-registration/label-selection-conus-medullaris.png
+   :align: center
+
+Repeat the same process for the posterior tip of the T9-T10 disc. Here, we will use the coordinate ``[22,77,187]``:
+
+.. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/lumbar-registration/label-selection-T9-T10-disc.png
+   :align: center
+
+The command below will open the input image in a GUI viewer, allowing you to select points to add labels to. In this case, we will add labels for the T9-T10 disc (numerical ID 17) and the conus medullaris (numerical ID 60).
```
Comment (sandrinebedard) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst:57]: There is no GUI since it direclty creates the labels at the coordinates we specified

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+   sct_deepseg -i t2_crop.nii.gz -task seg_lumbar_sc_t2w -qc ~/qc_singleSubj
+
+:Input arguments:
+   - ``-i`` : Input image
+   - ``-task`` : The eep learning segmentation task to apply to the image. In this case, we wa
+   - ``-qc`` : Directory for Quality Control reporting. QC reports allow us to evaluate the segmentation slice-by-slice
+
+:Output files/folders:
+   - ``t2_seg.nii.gz`` : 3D binary mask of the segmented spinal cord
```
Comment (sandrinebedard) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst:62]: My output name is `t2_crop_seg-manual.nii.gz` and not `t2_crop_seg.nii.gz`

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+:Input arguments:
+   - ``-i`` : Input image
+   - ``-s`` : Segmented spinal cord corresponding to the input image
```
Comment (sandrinebedard) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:18]:
```suggestion
   - ``-s`` : Segmented spinal cord corresponding to the input image.
```

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst [hunk 7]
```diff
@@ -0,0 +1,36 @@
+.. _applying-lumbar-registration-algorithm:
+
+Applying the registration algorithm
+###################################
+
+To apply the registration algorithm, the following command is used:
+
+.. code:: sh
+
+   sct_register_to_template -i t2_crop.nii.gz \
```
Comment (sandrinebedard) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:10]: To I need to go to a specific branch to get the updated PAM50 with the 60 label?

**Review by sandrinebedard — COMMENTED** [review]
Thank you @joshuacwnewton for this awsome tutorial!! I left some review, mainly  small nitpicks

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration.rst [hunk 8]
```diff
@@ -0,0 +1,14 @@
+.. _lumbar-registration:
+
+Registering lumbar images to the PAM50 template
+###############################################
+
+This tutorial demonstrates how to `register <http://jpeelle.net/mri/image_processing/registration.html>`_ a lumbar anatomical MRI scan to the PAM50 Template. T2 images are used for this tutorial, due to lumbar segmentation only being available for the T2 contrast.
```
Comment (valosekj) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration.rst:6]:
```suggestion
This tutorial demonstrates how to `register <http://jpeelle.net/mri/image_processing/registration.html>`_ a lumbar anatomical MRI scan to the PAM50 template. T2 images are used for this tutorial, due to lumbar segmentation only being available for the T2 contrast.
```

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst (hunk 3 above)
Comment (valosekj) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst:6]:
```suggestion
   * ``single_subject/data/t2_lumbar/t2.nii.gz`` : T2w anatomical scan of the lumbar spinal cord.
```

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+In this case, the lumbar region occupies the lowermost 100 axial slices of the image. But, we want to include a bit of the thoracic region for registration purposes, so we keep an extra 100 slices corresponding to the T9-T12 region.
+
+.. code:: sh
+
+   sct_crop_image -i t2.nii.gz -zmax 200
+
+:Input arguments:
+   - ``-i`` : Input image
+   - ``-zmax`` : The maximum z slice to keep. For an RPI image, the z axis corresponds to the axial plane, so specifying 200 means "keep the axial slices from 0-200".
```
Comment (valosekj) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst:23]:
```suggestion
   - ``-zmax`` : The maximum z slice to keep. For an image with RPI orientation, the z axis corresponds to the axial plane, so specifying 200 means "keep the axial slices from 0-200".
```

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst (hunk 4 above)
Comment (valosekj) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst:57]: We can maybe add label 60 to our [Labeling conventions](https://spinalcordtoolbox.com/user_section/tutorials/registration-to-template/vertebral-labeling/labeling-conventions.html).

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+To correct for this, the PAM50 template provides a cauda equinea label (specifically, the conus medullaris) as a registration landmark. By creating a similar label in your subject data, you can align the terminal end of the spinal cord between your subject and the PAM50 template. However, this comes with the necessary tradeoff of potentially misaligning the discs.
+
+Deciding which landmarks to use (discs, conus medullaris)
+=========================================================
+
+Registration can be performed with either one or two labels. So, given the tradeoffs between the aligning the discs and/or the conus medullaris, the question becomes: Should registration be done with 1 disc label, 1 conus medullaris label, or a combination of the two?
+
+.. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/lumbar-registration/label-comparison.png
```
Comment (valosekj) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst:17]: Nice figure!

Two minor ideas:
- we could add the `17` and `60` labels
- we could add green and red arrows highlight pros and cons areas

image

Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst:17]: Ooh, these additions are absolutely gorgeous! Thank you for adding them.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst (hunk 5 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst:62]: Fixed by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4235! :tada:

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst:6]: This suggestion overlaps with @valosekj's suggestion https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4250#discussion_r1355031504, so I'll combine the two suggestions and mark this as resolved.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst (hunk 7 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:10]: Fixed by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4250/commits/1dabb41e6cba1dfda42b18b6a6a95e7ab5fcc423.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+   You can get these files by downloading :sct_tutorial_data:`data_lumbar-registration.zip`.
```
[the remark this answers is not in the document]
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst:8]: Fixed by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4250/commits/0062f397e81c2a87ff35f342905598e2fdb51470.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst (hunk 4 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst]: Good idea!

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+:Input arguments:
+   - ``-i`` : Input image
+   - ``-task`` : The eep learning segmentation task to apply to the image. In this case, we wa
+
+:Output files/folders:
+   - ``t2_seg.nii.gz`` : 3D binary mask of the segmented spinal cord
+
+Once the command has finished, at the bottom of your terminal there will be instructions for inspecting the results using :ref:`Quality Control (QC) <qc>` reports.
+
+.. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/lumbar-registration/io_cropping-and-segmentation.png
```
Comment (NathanMolinier) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst:65]: I personally like the full image of the subject, however, I think the input of the function `sct_deepseg` is only the cropped image.

Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst:65]: Ah, good point! I should have two separate figures here: One for the cropping, and one for the segmentation.

Comment (NathanMolinier) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst:65]: Ooh, maybe I did not understand the purpose of the figure ! The input actually represents the data at the beginning of the page and the output the data at the end.

Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst:65]: Yes, exactly! But, if it was unclear for you on first read, then it will probably be unclear for our users, too. So, having two separate figures will definitely help clear things up. :)

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst (hunk 4 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst]: Done!

File: spinalcordtoolbox/download.py [hunk 13]
```diff
@@ -46,8 +46,9 @@
     },
     "PAM50": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20230519/PAM50-r20230519.zip",
-            "https://osf.io/8cgme/?action=download",
+            # DO NOT MERGE. This is a prerelease. This should be superseded by a new PAM50 link once spinal level changes are merged.
+            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20231011/PAM50-r20231011.zip",
+            "https://osf.io/dj93u/?action=download",
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/download.py:51]: It looks like adding the label `60` caused one of our existing tests to fail:

```
test_sct_label_vertebrae_high_value_warning - AssertionError: assert 'Disc value not included in template.' in 'WARNING  
```
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/78d46e0d1a75dfa6a27689931bf4111a233c04f1/testing/cli/test_cli_sct_label_vertebrae.py#L70-L75

What _should_ happen is that `-initz 40,19` (z=40, label=19) should trigger this warning:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/78d46e0d1a75dfa6a27689931bf4111a233c04f1/spinalcordtoolbox/vertebrae/core.py#L208-L212

(This happens because we start with `19`, then increment in the inferior direction (`20`), and then `list_disc_value_template.index(current_value)` fails with `current_value==20`)

I will try to debug why the addition of the `60` label causes this to _not_ happen.

Comment (joshuacwnewton) [reply spinalcordtoolbox/download.py:51]: Issue opened here: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4254

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst (hunk 12 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/lumbar-segmentation.rst]: Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4250/commits/65c1fa0615543db4417bc059527b8918ac5358eb. :)

File: spinalcordtoolbox/download.py (hunk 13 above)
Comment (joshuacwnewton) [reply spinalcordtoolbox/download.py:51]: Resolved by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4250/commits/de1019ee67d94e829160a28210894ed3636afd1f and https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4250/commits/0187e613cfddaa2bf7f1cc2ae32c11988d7bc8ee.

For justifications for these changes, see https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4254#issuecomment-1758581360.

File: spinalcordtoolbox/download.py [hunk 14]
```diff
@@ -46,8 +46,9 @@
     },
     "PAM50": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20230519/PAM50-r20230519.zip",
-            "https://osf.io/8cgme/?action=download",
+            # DO NOT MERGE. This is a prerelease. This should be superseded by a new PAM50 link once spinal level changes are merged.
+            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20231011/PAM50-r20231011.zip",
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/download.py:50]: Note to self: The PAM50 template will be properly updated in a separate PR. We need to make sure _that_ URL is also added to this PR, rather than the temporary prerelease I created above.

File: spinalcordtoolbox/download.py (hunk 13 above)
Comment (joshuacwnewton) [reply spinalcordtoolbox/download.py:51]: These changes have been migrated to https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4268.

File: spinalcordtoolbox/download.py (hunk 14 above)
Comment (joshuacwnewton) [reply spinalcordtoolbox/download.py:50]: As soon as https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4268 is merged, I will revert the PAM50 related changes from this PR.

**Comment by mguaypaq** [comment]
I went through the tutorial, and only ran into one problem: because I had an earlier release of the PAM50 template, I got the following error message for the `sct_register_to_template` command:
```
ERROR: Wrong landmarks input. Labels must have correspondence in template space. 
Label max provided: 60
Label max from template: 21
```
After getting the latest release with `sct_download_data -d PAM50`, the command worked. Since some users are likely to run into the same problem, I would suggest adding a note about this (both the error message, and how to fix it).

Also, this is a very minor point, and probably not worth fixing, but the automatic syntax highlighting for shell commands is a little weird. Specifically, it only highlights the first number for the `sct_label_utils` command. (I would say that highlighting numbers in general is a little weird, but this command is worse than the others.)

**Comment by jcohenadad** [comment]
> Since some users are likely to run into the same problem, I would suggest adding a note about this (both the error message, and how to fix it).

Yes, I very much agree with this. It is very common that users attend the course with a non-up-to-date SCT version (eg: some are currently processing data and they don't want to upgrade SCT)

> Also, this is a very minor point, and probably not worth fixing, but the automatic syntax highlighting for shell commands is a little weird. Specifically, it only highlights the first number for the sct_label_utils command. (I would say that highlighting numbers in general is a little weird, but this command is worse than the others.)

what highlighting are we talking about? is that on the gslides?

**Comment by mguaypaq** [comment]
> > Also, this is a very minor point, and probably not worth fixing, but the automatic syntax highlighting for shell commands is a little weird. Specifically, it only highlights the first number for the sct_label_utils command. (I would say that highlighting numbers in general is a little weird, but this command is worse than the others.)
> 
> what highlighting are we talking about? is that on the gslides?

No, I meant in the new tutorial, on https://spinalcordtoolbox.com. For example, [the shell command at this link](https://spinalcordtoolbox--4250.org.readthedocs.build/user_section/tutorials/registration-to-template/lumbar-registration/labeling.html#codecell0), highlights the number 22 in bold blue, but not the other numbers:

image

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst (hunk 1 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst:20]: @jcohenadad suggests that the 60 label is hard to see

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst (hunk 2 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:14]: Feedback given in SCT meeting -- will repeat here later.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+                            -s t2_crop_seg.nii.gz \
+                            -ldisc t2_crop_labels.nii.gz \
+                            -c t2 -qc ~/qc_singleSubj \
+                            -param step=1,type=seg,algo=centermassrot:step=2,type=seg,algo=bsplinesyn,metric=MeanSquares,iter=3,slicewise=0:step=3,type=im,algo=syn,metric=CC,iter=3,slicewise=0
+
```
Comment (mguaypaq) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:15]: The instructions for updating the PAM50 template are four pages away, and easy to overlook. I would suggest adding the exact error message here, close to where it's likely to happen, with a link to the instructions.

```suggestion

.. note::

   If you get the following error message, it means that your installed version of the PAM50 template is too old:
   
   .. code::
   
      ERROR: Wrong landmarks input. Labels must have correspondence in template space. 
      Label max provided: 60
      Label max from template: 21   

   In that case, you should update it by following `these instructions <before-starting-lumbar-registration>`_.
```

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst [hunk 16]
```diff
@@ -0,0 +1,22 @@
+Before starting this tutorial
```
Comment (mguaypaq) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/before-starting.rst:1]: (An anchor for the link in my other suggestion.)

```suggestion
.. _before-starting-lumbar-registration:

Before starting this tutorial
```

**Review by mguaypaq — APPROVED** [review]
Maybe it's too much, but I suggest stating the explicit error message for outdated PAM50 templates. Feel free to decline. Other than that, the tutorial works, so I'll pre-approve the pull request. You can merge it once the remaining conversations are resolved.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst (hunk 1 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst:20]: I've made a handful of tweaks to improve readability:

Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/labeling.rst:20]: I've made a handful of tweaks here to improve readability:

image

- Higher contrast label colors
- Larger label text size
- Higher contrast arrows
- Using identical arrow locations for all 3 pictures (to make comparing/contrasting easier)
- Left-aligning the bottom text
- Using emojis to better clarify intent

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst (hunk 15 above)
Comment (joshuacwnewton) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:15]: Lovely suggestion! I very much think this is a good idea.

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+   If you get the following error message, it means that your installed version of the PAM50 template is too old:
+   
+   .. code::
+   
+      ERROR: Wrong landmarks input. Labels must have correspondence in template space. 
+      Label max provided: 60
+      Label max from template: 21   
+
+   In that case, you should update it by following `these instructions <before-starting-lumbar-registration>`_.
+:Input arguments:
```
Comment (joshuacwnewton) [inline documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:27]: Fix RTD error:

```
lumbar-registration/template-registration.rst:27: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

```suggestion
   In that case, you should update it by following `these instructions <before-starting-lumbar-registration>`_.

:Input arguments:
```

**Comment by joshuacwnewton** [comment]
I think this is pretty much ready to be merged? 

@jcohenadad is just going over the SCT Course slides for lumbar registration (see [this comment](https://docs.google.com/presentation/d/1t40Fd0fS0SwWR5FU_GWKEvHkB9d96LVddLQW6L3QAx0/edit?disco=AAAA_O14ye0)), so I will wait to see if any new feedback comes from that review. If everything looks good, then I will merge. :)

File: documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst (hunk 2 above)
Comment (jcohenadad) [reply documentation/source/user_section/tutorials/registration-to-template/lumbar-registration/template-registration.rst:14]: the syn step is WAAAY better! thank you for tweaking this!

**Comment by joshuacwnewton** [comment]
I've updated this tutorial (as well as the `sct_tutorial_data` repo) with the changes suggested by Julien (namely, removing the full body `t2.nii.gz` scan and replacing it with a `t2_lumbar.nii.gz` cropped scan).

I've named the new release [SCT-Course-20231020](https://github.com/spinalcordtoolbox/sct_tutorial_data/releases/tag/SCT-Course-20231020) to match existing conventions for past SCT courses, to note which release was actually used in the SCT Course.

I'm going to do one more quick run-through of the tutorial myself, but otherwise I think this is good to merge.

**Comment by joshuacwnewton** [comment]
Thank you to everyone who tested and reviewed this PR! :hearts:
